File: scripts/parse_test.py
import sys
from lib.pdf_utils import extract_pdf_lines, extract_doc_meta
from lib.chunker import chunk_document


def main():
    path = sys.argv[1]
    meta = extract_doc_meta(path)
    lines = extract_pdf_lines(path)
    chunks = chunk_document(lines)

    print("=== META ===", meta)
    print(f"=== CHUNKS ({len(chunks)}) ===")
    for c in chunks:
        print(f"\n[{c['section']}] p.{c['page']}")
        print(c["text"][:300])


if __name__ == "__main__":
    main()

File: scripts/parse-test.ts
import fs from "fs";
import path from "path";
import { extractPdfLines, extractDocMeta } from "./lib/pdf";
import { chunkDocument } from "./lib/chunker";

async function main() {
  const file = process.argv[2];
  const buffer = fs.readFileSync(file);
  const meta = await extractDocMeta(buffer);
  const { lines, numPages } = await extractPdfLines(buffer);
  const chunks = chunkDocument(lines);

  console.log("=== META ===", meta, "pages:", numPages);
  console.log("=== CHUNKS (%d) ===", chunks.length);
  for (const c of chunks) {
    console.log(`\n[${c.section}] p.${c.page}`);
    console.log(c.text.slice(0, 300));
  }
}

main();
